demo/lazyload-demo.html

```diff
@@ -0,0 +1,152 @@
+                    //Get the grandchildren for this child.
+                    self.getChildren(child.children).then(function(childrenArr){
+                        self.addChildrenToTree(childrenArr, updatePath);
+                    });
+                });
+            },
+            ready: function(){
+                var self = this;
+                // Get the root doc for the tree
+                self.getDoc(this.rootDocId).then(function(rootDoc){
+                    var result = {
+                        name: rootDoc.name,
+                        open: false,
+                        updatePath: 'doc',
+                        loading: true,
+                        children: rootDoc.children
+                    };
+                    self.doc = result;
+                    //Get the children for the root.
+                    self.getChildren(rootDoc.children).then(function(childrenArr){
+                        self.addChildrenToTree(childrenArr, 'doc');
+                    });
+                });
+            },
+            addChildrenToTree: function(childrenArr, updatePath){
+                var self = this;
+                var idx = 0;
+                var results = [];
+                childrenArr.forEach(function(child){
+                    results.push({
+                        name: child.name,
+                        open: false,
+                        updatePath: updatePath+'.children.'+idx,
+                        loading: true,
+                        children: child.children
+                    });
+                    idx++;
+                });
+                self.set(updatePath+'.loading', false);
+                self.set(updatePath+'.children', results);
+            },
+            getDoc: function(docId) {
+                //Pretend we're getting data from the data store
+                var self = this;
+                return new Promise(function(resolve, reject) {
+                    window.setTimeout(function() {
+                        resolve(self.data[docId]);
+                    }, 1000);
+                });
+            },
+            getChildren: function(childrenIdArr) {
+                var self = this;
+                //Pretend we're getting data from the data store
+                if(!childrenIdArr) return Promise.resolve([]);
+                return new Promise(function(resolve, reject) {
+                    window.setTimeout(function() {
+                        var resultsArr = [];
+                        childrenIdArr.forEach(function(childId){
+                            resultsArr.push(self.data[childId]);
+                        });
+                        resolve(resultsArr);
+                    }, Math.floor(Math.random() * 2000));
+                });
+            },
+            dataChanged: function(newValue, oldValue) {
+                console.log('doc:', newValue);
+            }
+
+        });
+    </script>
+</dom-module>
+
```

Comment:
could you add it as a dom bind (instead of a component) in the demo page. and fix the former demos BTW  😅